doc/install/manual-deployment.rst

```diff
@@ -348,116 +348,70 @@ Long Form
-   OSD starts up. The following command will output the OSD number, which you 
-   will need for subsequent steps. ::
-	
-	ceph osd create [{uuid} [{id}]]
-
-
-#. Create the default directory on your new OSD. :: 
-
-	ssh {new-osd-host}
-	sudo mkdir /var/lib/ceph/osd/{cluster-name}-{osd-number}
-	
+#. Generate a UUID for the OSD. ::
 
-#. If the OSD is for a drive other than the OS drive, prepare it 
-   for use with Ceph, and mount it to the directory you just created:: 
+     UUID=$(uuidgen)
 
-	ssh {new-osd-host}
-	sudo mkfs -t {fstype} /dev/{hdd}
-	sudo mount -o user_xattr /dev/{hdd} /var/lib/ceph/osd/{cluster-name}-{osd-number}
+#. Generate a cephx key for the OSD. ::
 
-	
-#. Initialize the OSD data directory. :: 
+     OSD_SECRET=$(ceph-authtool --gen-print-key)
 
-	ssh {new-osd-host}
-	sudo ceph-osd -i {osd-num} --mkfs --mkkey --osd-uuid [{uuid}]
+#. Create the OSD. Note that an OSD ID can be provided as an
+   additional argument to ``ceph osd new`` if you need to reuse a
+   previously-destroyed OSD id. We assume that the
+   ``client.bootstrap-osd`` key is present on the machine.  You may
+   alternatively execute this command as ``client.admin`` on a
+   different host where that key is present.::
 	
-   The directory must be empty before you can run ``ceph-osd`` with the 
-   ``--mkkey`` option. In addition, the ceph-osd tool requires specification
-   of custom cluster names with the ``--cluster`` option.
-
-
-#. Register the OSD authentication key. The value of ``ceph`` for 
-   ``ceph-{osd-num}`` in the path is the ``$cluster-$id``.  If your 
-   cluster name differs from ``ceph``, use your cluster name instead.::
+     ID=$(echo "{\"cephx_secret\": \"$OSD_SECRET\"}" | \
+	ceph osd new $UUID -i - \
+	-n client.bootstrap-osd -k /var/lib/ceph/bootstrap-osd/ceph.keyring)
 
-	sudo ceph auth add osd.{osd-num} osd 'allow *' mon 'allow profile osd' -i /var/lib/ceph/osd/{cluster-name}-{osd-num}/keyring
-
-
-#. Add your Ceph Node to the CRUSH map. ::
-
-	ceph [--cluster {cluster-name}] osd crush add-bucket {hostname} host
-
-   For example::
+#. Create the default directory on your new OSD. ::
 
-	ceph osd crush add-bucket node1 host
+     mkdir /var/lib/ceph/osd/ceph-$ID
 
+#. If the OSD is for a drive other than the OS drive, prepare it 
+   for use with Ceph, and mount it to the directory you just created. ::
 
-#. Place the Ceph Node under the root ``default``. ::
-
-	ceph osd crush move node1 root=default
+     mkfs.xfs /dev/{DEV}
+     mount /dev/{DEV} /var/lib/ceph/osd/ceph-$ID
 
+#. Write the secret to the OSD keyring file.::
 
-#. Add the OSD to the CRUSH map so that it can begin receiving data. You may
-   also decompile the CRUSH map, add the OSD to the device list, add the host as a
-   bucket (if it's not already in the CRUSH map), add the device as an item in the
-   host, assign it a weight, recompile it and set it. ::
+     printf "[osd.$ID]\n\tkey = $OSD_SECRET\n" > /var/lib/ceph/osd/ceph-$ID/keyring
```

Comment:
The way we are doing this is with `ceph-authtool`, like:

    ceph-authtool /var/lib/ceph/osd/ceph-$ID/keyring --create-keyring --name osd.$ID --add-key $OSD_SECRET

Is that wrong? are we just trying to exemplify the most manual way here?